### PR TITLE
Adjust the configureTaskUsingProjectProperty sample for configuration cache

### LINKED ISSUE #21825: CCH22-28: Adjust the `snippets/tutorial/configureTaskUsingProjectProperty` sample for configuration cache
The sample project located under [`subprojects/docs/src/snippets/tutorial/configureTaskUsingProjectProperty`](https://github.com/gradle/gradle/tree/master/subprojects/docs/src/snippets/tutorial/configureTaskUsingProjectProperty) does not meet [the configuration cache requirements](https://docs.gradle.org/nightly/userguide/configuration_cache.html#config_cache:requirements).

The build scripts in the `groovy` and `kotlin` parts of the sample project need to be adjusted to follow the requirements.
### Context
This issue is a part of the [Gradle Configuration Cache Hackathon 2022](https://github.com/gradle/cc-hackathon-2022).

If you wish to participate and contribute the fix, please follow [the suggested workflow](https://github.com/gradle/cc-hackathon-2022/blob/main/workflow.md).
### Expected behavior

- The sample project is fixed so that the related documentation tests pass with configuration cache enabled.
- The tests related to this sample project are removed from the `testsToBeFixedForConfigurationCache` list in `subprojects/docs/build.gradle`. You can find the test IDs below.
- The doc pages that include the snippets from this sample project might need to be updated to be consistent with the sample. Feel free to ask the team for help with this! You can find the pages listed below.
### Test failure details
These tests running in `:docs:docsTest` check the sample project and are failing with configuration cache enabled:
* `snippet-tutorial-configure-task-using-project-property_groovy_configureTaskUsingProjectProperty.sample`[(Build Scan)](https://ge.gradle.org/s/wkszl3ueqiln2/tests/:docs:docsTest/org.gradle.docs.samples.Bucket1SnippetsTest/snippet-tutorial-configure-task-using-project-property_groovy_configureTaskUsingProjectProperty.sample)
    ```
    Build file 'build.gradle': invocation of 'Task.project' at execution time is unsupported.
    ```

* `snippet-tutorial-configure-task-using-project-property_kotlin_configureTaskUsingProjectProperty.sample`[(Build Scan)](https://ge.gradle.org/s/wkszl3ueqiln2/tests/:docs:docsTest/org.gradle.docs.samples.Bucket1SnippetsTest/snippet-tutorial-configure-task-using-project-property_kotlin_configureTaskUsingProjectProperty.sample)
    ```
    Task `:performRelease` of type `org.gradle.api.DefaultTask`: invocation of 'Task.project' at execution time is unsupported.
    ```

### How to reproduce
To reproduce the failure in the Gradle documentation tests, run the following in the `gradle/gradle`repository:
```sh
./gradlew :docs:docsTest \
    -PrunBrokenConfigurationCacheDocsTests=true \
    -PenableConfigurationCacheForDocsTests=true \
    --tests "*snippet-tutorial-configure-task-using-project-property_groovy_configureTaskUsingProjectProperty.sample" \
    --tests "*snippet-tutorial-configure-task-using-project-property_kotlin_configureTaskUsingProjectProperty.sample"
```
In the test build outputs, which you can find in the test report, the configuration cache problems are listed along with the URL pointing to the detailed HTML problems report that can help investigating the violation. 
You can find more instructions on working with the sample projects in the [FAQ](https://github.com/gradle/cc-hackathon-2022/blob/main/faq.md).
### Docs including the snippets from this sample project
The following doc pages include the snippets from this sample project: 
- `subprojects/docs/src/docs/userguide/running-builds/build_environment.adoc`;

### PR DESCRIPTION
**Pull request by SidB3**
Fixes #21825

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/docs/src/snippets/tutorial/configureTaskUsingProjectProperty/groovy/build.gradle [hunk 1]
```diff
@@ -1,9 +1,11 @@
 tasks.register('performRelease') {
+    def isCI = providers.gradleProperty("isCI")
     doLast {
-        if (project.hasProperty("isCI")) {
+        if (isCI.isPresent()) {
```
Comment (bamboo) [inline subprojects/docs/src/snippets/tutorial/configureTaskUsingProjectProperty/groovy/build.gradle:4]: 💄 Prefer Groovy property syntax `isCI.present`

File: subprojects/docs/src/snippets/tutorial/configureTaskUsingProjectProperty/kotlin/build.gradle.kts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 tasks.register("performRelease") {
+    val isCI = providers.gradleProperty("isCI")
     doLast {
-        if (project.hasProperty("isCI")) {
+        if (isCI.isPresent()) {
```
Comment (bamboo) [inline subprojects/docs/src/snippets/tutorial/configureTaskUsingProjectProperty/kotlin/build.gradle.kts:4]: 💄 Prefer Kotlin property syntax `isCI.isPresent` (no parenthesis)

**Review by bamboo — CHANGES_REQUESTED** [review]

**Comment by SidB3** [comment]
Thanks @bamboo . PR updated.

File: subprojects/docs/src/snippets/tutorial/configureTaskUsingProjectProperty/groovy/build.gradle [hunk 3]
```diff
@@ -1,9 +1,11 @@
 tasks.register('performRelease') {
+    def isCI = providers.gradleProperty("isCI")
     doLast {
-        if (project.hasProperty("isCI")) {
+        if (isCI.isPresent) {
```
Comment (bamboo) [inline subprojects/docs/src/snippets/tutorial/configureTaskUsingProjectProperty/groovy/build.gradle:4]: 🔴 In Groovy, the idiomatic property name would be just `present`.

**Review by bamboo — CHANGES_REQUESTED** [review]

**Comment by bamboo** [comment]
This is looking good, @SidB3 👍 

Please add your full name and github handle to the list of contributors in the release notes as described [here](https://github.com/gradle/gradle/blob/master/subprojects/docs/src/docs/release/notes.md?plain=1#L6).

**Comment by bamboo** [comment]
@bot-gradle test this

**Comment by bamboo** [comment]
@bot-gradle test this

**Review by bamboo — APPROVED** [review]
🎉

**Comment by bamboo** [comment]
@bot-gradle test and merge

**Comment by bamboo** [comment]
@bot-gradle test and merge

**Pull request merged**